src/components/pages/group/edit/homepage-customization/gallery-setup/gallery-setup-table/index.tsx

```diff
@@ -0,0 +1,84 @@
+import { Button, ButtonGroup } from "@chakra-ui/core";
+import useTranslation from "@configs/i18n/useTranslation";
+import { axRemoveCustomField, axReorderCustomField } from "@services/usergroup.service";
+import notification, { NotificationType } from "@utils/notification";
+import arrayMove from "array-move";
+import React, { useEffect, useState } from "react";
+
+import GalleryListItems from "./gallery-list";
+
+const GallerySetupTable = ({ userGroupId, galleryList, setGalleryList, setIsCreate }) => {
+  const [canReorder, setCanReorder] = useState<boolean>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    setGalleryList(galleryList.sort((a, b) => a.displayOrder - b.displayOrder));
+  }, []);
+
+  const onSortEnd = ({ oldIndex, newIndex }) => {
+    setGalleryList(arrayMove(galleryList, oldIndex, newIndex));
+    setCanReorder(true);
+  };
+
+  const handleReorderCustomField = async () => {
+    const payload = galleryList.map((field, displayOrder) => ({
+      cfId: field.galleryList.id,
+      displayOrder
+    }));
+
+    const { success } = await axReorderCustomField(userGroupId, payload);
+    if (success) {
+      notification(t("GROUP.HOMEPAGE_CUSTOMIZATION.REORDER.SUCCESS"), NotificationType.Success);
+    } else {
+      notification(t("GROUP.HOMEPAGE_CUSTOMIZATION.REORDER.FAILURE"));
+    }
+    setCanReorder(false);
+  };
+
+  const removeGalleryItem = async (index) => {
+    const { success } = await axRemoveCustomField(userGroupId, index);
+    if (success) {
+      setGalleryList(galleryList.filter(({ galleryList: { id } }) => id !== index));
```

Comment:
No need of unnecessary destructiring here just simply use something like
```suggestion
      setGalleryList(galleryList.filter(gallery) => gallery.gallerylist.id !== index));
```
this is much cleaner to understand and destructiring has a cost so preferabally we can avoid wherever we can